tributors/main/orcid.py

```diff
@@ -132,77 +130,119 @@ def get_orcid_token():
+        )
 
-def get_orcid(email, token, name=None):
-    """Given a GitHub repository address, retrieve a lookup of contributors
-       from the API endpoint. We look to use the GITHUB_TOKEN if exported
-       to the environment, and exit if the response has any issue
+        if not choice or choice in ["s", "S", "skip"]:
+            return
+
+        # Return the orcid identifier
+        return results[int(choice) - 1]["orcid-id"]
+
+
+def get_orcid(email, name=None, interactive=False):
+    """Get an orcid identifier for a given email or name.
     """
-    # We must have a token, and an email OR name
-    if not token and (not email and not name):
+    # We must have an email OR name
+    if not email and not name:
         return
 
     # First look for records based on email
     orcid_id = None
     if email:
-        url = "https://pub.orcid.org/v3.0/search/?q=email:%s" % email
-        orcid_id = record_search(url, token, email)
+        url = "https://pub.orcid.org/v3.0/expanded-search?q=email:%s" % email
+        orcid_id = record_search(url, email, interactive)
 
     # Attempt # 2 will use the first and last name
     if name is not None and not orcid_id:
 
-        # If no comma, likely a space delimiter
         delim = "," if "," in name else " "
         cleaner = "," if delim == " " else " "
 
         parts = name.split(delim)
+
+        # No go if only a first or last name
+        if len(parts) == 1:
+            bot.debug(f"Skipping {name}, first and last are required for search.")
+            return orcid_id
+
         last, first = parts[0].strip(cleaner), " ".join(parts[1:]).strip(cleaner)
-        url = (
-            "https://pub.orcid.org/v3.0/search/?q=given-names:%s+AND+family-name:%s"
-            % (first, last)
-        )
-        orcid_id = record_search(url, token, name)
+        url = "https://pub.orcid.org/v3.0/expanded-search?q=%s+AND+%s" % (first, last)
+        orcid_id = record_search(url, name, interactive)
 
         # Attempt # 3 will try removing the middle name
         if " " in first:
-            url = (
-                "https://pub.orcid.org/v3.0/search/?q=given-names:%s+AND+family-name:%s"
-                % (first.split(" ")[0].strip(), last)
+            url = "https://pub.orcid.org/v3.0/expanded-search?q=%s+AND+%s" % (
+                first.split(" ")[0].strip(),
+                last,
             )
-            orcid_id = record_search(url, token, name)
+            orcid_id = record_search(url, name, interactive)
 
-        # Attempt # 4 will reverse (some orcid entries are last, first, others the opposite
+        # Last attempt tries full name
         if orcid_id is None:
-            url = (
-                "https://pub.orcid.org/v3.0/search/?q=given-names:%s+AND+family-name:%s"
-                % (last, first)
-            )
-            orcid_id = record_search(url, token, name)
+            url = "https://pub.orcid.org/v3.0/expanded-search?q=%s" % (name)
```

Comment:
nit: the same here as below about `(name)` not being an (expected to be?) tuple...